### PR TITLE
feat: docs for web-wallet

### PR DESCRIPTION
**Pull request by 0xpulkit**

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -162,6 +162,7 @@ module.exports = {
 
       ],
     },
+    "web-wallet",
```
Comment (amonsosanz) [inline sidebars.js:165]: We should keep the options on both menus synced (except Introduction, Quick Start Demo and Releases):

image

This means:

* Adding Web Wallet also to the top menu
* Moving JS-SDK below smart contracts in the top menu (or the other way around in the side menu)

Comment (0xpulkit) [reply sidebars.js:165]: Yes i had written this down for discussion cauz we need to decide on whether to have web wallet in 'Tools' or 'Build' technically as it is a tool it should be placed in tools. The same was suggested by @elias-garcia as well.
But i noticed that all the tools are pointing to an external url where people can use the tool, the same is not possible with Web Wallet- shoud we point it to its documentation istelf?

File: docs/web-wallet.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+- **Simplified Integration**: Integration into your application is effortless, reducing the development workload and ensuring swift implementation.
+- **On-Chain and Off-Chain Query Verification**: Supporting both on-chain and off-chain query verification, the Web Wallet offers flexibility to meet diverse verification needs.
+- **Consistency and Accuracy**: With a standardized interface, the tool promotes uniformity across users and organizations, minimizing errors.
+
+## Getting Started
+
+Learn more about the Web Wallet in the video below:
+
+<div align="center" style={{margin: 40}}>
+<iframe src="https://www.youtube.com/embed/" width="100%" height="450" allowfullscreen></iframe>
```
Comment (amonsosanz) [inline docs/web-wallet.md:53]: Since `src` is not yet filled, I would mark this pull request as a draft so we don't forget and merge it by mistake.

File: docs/web-wallet.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+## Integration Process for the Verifiers
+
+### Step 1 - Choosing Your Credential Schema
+
+Select the credential schema that aligns with your requirements from our [Schema Explorer](https://schema-builder.polygonid.me/) or any other source.
+Standardized schemas, such as KYC, are available. Alternatively, you can create a custom schema using our [Schema Builder](https://schema-builder.polygonid.me/builder).
+
+### Step 2 - Creating the Query
+
+Define what information you need from your users and build a query using the Query Builder [https://schema-builder.polygonid.me/query-builder ].
```
Comment (amonsosanz) [inline docs/web-wallet.md:89]: This link fails to render due to the last space.

File: docs/web-wallet.md (hunk 2 above)
Comment (0xpulkit) [reply docs/web-wallet.md:55]: Sure , makes sense @amonsosanz

Comment (0xpulkit) [reply docs/web-wallet.md]: Have removed it for now so that this goes to staging- i will add the link when ready

File: docs/web-wallet.md (hunk 3 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: fixed

File: sidebars.js (hunk 1 above)
Comment (amonsosanz) [reply sidebars.js:165]: Well, these are different discussions. My previous comment applies to whatever we decide to do regarding tools. In Build, we need to point to the same documentation page we point from the sidebar.

> the same is not possible with Web Wallet

Why is it not possible? I think of the demo app as the entry point to the wallet.

Comment (0xpulkit) [reply sidebars.js:165]: Got it @amonsosanz , I will do as follows:
1) sync the content in build and sidebar
2) Add wallet web in tools and point it to the demo

File: docs/web-wallet.md (hunk 2 above)
Comment (amonsosanz) [reply docs/web-wallet.md]: But this is not JS so `//add yt video` is rendered...

File: docs/web-wallet.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+3. **Creating Identity and Accessing Credentials**: Next, the user selects 'Sign Message' to create a derived identity wallet and retrieve the encrypted credentials associated with their Ethereum wallet via the cloud storage.
+4. **Claiming the Credential**: If the user hasn't previously claimed the credential required by the verifier’s request, they proceed by clicking 'Claim Credential' and complete the issuance process (a face scan in this scenario) to claim the credential via the embedded issuance.
+5. **Proof generation**: Once the user owns the credential, can select 'Verify', triggering the generation of the zk-proof which is then shared to the verifier application in case of off-chain verification and to the smart contract in case of on-chain verification
+6. **Verification Completion**: The user finally clicks the ‘Continue’ button, upon which they are redirected to the application where the verification can be completed and the application flow can continue.
+
+### End-User Journey of Verification of other credentials
+
+When proving custom authentication on an application:
```
Comment (amonsosanz) [inline docs/web-wallet.md:70]: This "proving custom authentication" sounds weird to me. I'm not even sure what "proving a custom authentication" is. I'd consider changing this to:

> When verifying credentials lacking support for embedded issuance on an application:

In any case, I have problems with this whole section. It repeats the previous steps but explains the option “Verify from mobile”, which should be also explained in the previous section.

I would consider completely removing this section and explaining the option “Verify from mobile” in the previous section. Then, we can add an info box below explaining this custom credential scenario:

ℹ️ For credentials that do not support embedded issuance directly through the Web Wallet, applications can still facilitate a seamless verification process. Users who have previously issued credentials stored on their mobile devices can continue the verification from their mobile device. To do this, users can simply click the “Verify from Mobile” button in the Web Wallet interface. This action seamlessly integrates with the existing credentials on their device, allowing them to continue with the verification flow without interruption.

Comment (0xpulkit) [reply docs/web-wallet.md]: Fixed

File: docs/web-wallet.md (hunk 2 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: Fixed

File: sidebars.js (hunk 1 above)
Comment (0xpulkit) [reply sidebars.js:165]: Done

Comment (amonsosanz) [reply sidebars.js:165]: I'm still missing the option Web Wallet in the Build top menu.

File: docs/web-wallet.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+:::
+
+- **Enhanced User Experience**: Prioritizing simplicity, the Web Wallet presents a user-friendly interface. From identity creation to credential claiming and verification, all steps seamlessly integrate, minimizing complexity.
+- **Embedded Issuance**: The Web Wallet provides embedded issuance, allowing users to claim their credentials within the verification flow if they haven't done so before.
+
+:::note
+
+Currently, the tool supports embedded issuance for only two types of credentials:
+
+- Proof of Identity (refer to the schema [here](https://github.com/anima-protocol/claims-polygonid/blob/main/schemas/json-ld/poi-v1.json-ld))
```
Comment (amonsosanz) [inline docs/web-wallet.md:40]: Links in boxes should use the `ins` tag so they are properly underlined.

`[<ins>here</ins>](https://github.com/anima-protocol/claims-polygonid/blob/main/schemas/json-ld/poi-v1.json-ld)`

File: sidebars.js (hunk 1 above)
Comment (0xpulkit) [reply sidebars.js:165]: Have added it in the Tools Top menu

File: docs/web-wallet.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+:::note
+
+The server setup for Off-Chain Verification and the Smart Contract setup for On-Chain Verification, as detailed in the provided links, remain unchanged. The only modification required is in the client side of the application. Previously, proof requests could only be shared via embedded QR codes. Now, verifiers have a new, more efficient method for requesting proofs from users by leveraging the Web Wallet.
+
+:::
+
+:::note
+
+In Case of Off-Chain Verification, once the user goes through the flow of verification in the Web Wallet, the zero-knowledge proof is returned as a JWZ token in a callback to your verifier backend. You need to handle this callback coming from the Web Wallet and verify the proof.
```
Comment (amonsosanz) [inline docs/web-wallet.md:105]: We could link to the JWZ docs [here](http://localhost:3000/docs/wallet/wallet-sdk/polygonid-sdk/iden3comm/jwz).

File: docs/web-wallet.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+  ],
+  callbackUrl: "https://my-app.org/api/callback",
+  verifierDid: "did:polygonid:polygon:amoy:2qV9QXdhXXmN5sKjN1YueMjxgRbnJcEGK2kGpvk3cq",
+};
+
+// Encode the verification request to base64
+const base64EncodedVerificationRequest = btoa(JSON.stringify(verificationRequest));
+
+// Open the Polygon ID Verification Web Wallet with the encoded verification request
+window.open(`https://web-wallet.polygonid.me${base64EncodedVerificationRequest}`);
```
Comment (amonsosanz) [inline docs/web-wallet.md:173]: This code won't work. You're lacking the anchor (#) in the template literal.

File: docs/web-wallet.md [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+    functionName: "submitZKPResponse",
+    chainId: 80002,
+    network: "polygon-amoy",
+  },
+};
+// Encode the verification request to base64
+const base64EncodedVerificationRequest = btoa(JSON.stringify(verificationRequest));
+
+// Open the Polygon ID Verification Web Wallet with the encoded verification request
+window.open(`https://web-wallet.polygonid.me${base64EncodedVerificationRequest}`);
```
Comment (amonsosanz) [inline docs/web-wallet.md:215]: This code won't work. You're lacking the anchor (#) in the template literal.

File: docs/web-wallet.md (hunk 5 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: Updated

File: docs/web-wallet.md [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
```
Comment (amonsosanz) [inline docs/web-wallet.md:215]: `transactionData` points to Amoy but the verification points to the production wallet. Since the contract address is correct, I'd leave this `transactionData` but point to the testing wallet instead, so the snippet works.

File: docs/web-wallet.md [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+```
+
+</TabItem>
+</Tabs>
+
+`backUrl` represents the URL endpoint of your application where the user will be redirected when they click the ‘Back’ button
+
+`finishUrl` represents the URL endpoint of your application where the user will be redirected once the proof has been successfully generated and, they click the ‘Continue’ button
```
Comment (amonsosanz) [inline docs/web-wallet.md:223]: I'd not use the word "endpoint" for the `backUrl` and the `finishUrl` since it suggests these URLs point to APIs while they are web pages.

File: docs/web-wallet.md [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+
+`name` represents the name of your Application.
+
+`zkQueries` represents the query for what you need to authenticate. Input the generated code snippet of the query in Step 2.
```
Comment (amonsosanz) [inline docs/web-wallet.md:227]: 2 observations here:

 * I'd rephrase "represents the query for what you need to authenticate" to "represents the query you request your users to verify"
 * I would emphasize that regardless of the plural in `zkQueries` and this property being a list, currently we only support one query per verification request. This is though prepared for the future support of multiple queries.

File: docs/web-wallet.md [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+
+`verifierDiD` represents the DiD of the verifier.
```
Comment (amonsosanz) [inline docs/web-wallet.md:229]: represents the DiD of the verifier (your DID)

File: docs/web-wallet.md [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+  backUrl: "https://my-app.org/back",
+  finishUrl: "https://my-app.org/finish",
+  logoUrl: "https://my-app.org/logo.png",
+  name: "My app",
+  zkQueries: [
+    {
+      circuitId: "credentialAtomicQuerySigV2",
+      id: 1711399135,
+      query: {
+        allowedIssuers: ["*"],
```
Comment (amonsosanz) [inline docs/web-wallet.md:192]: We need to add a warning/alert box clarifying that this value needs to be populated with the DIDs of actual issuers you trust.

Similar to what we do [here](http://localhost:3000/docs/verifier/verification-library/zk-query-language/#multi-query) and [here](http://localhost:3000/docs/verifier/verification-library/verifier-setup/#verifier-server-setup).

File: docs/web-wallet.md (hunk 11 above)
Comment (amonsosanz) [reply docs/web-wallet.md:227]: Ah sorry the box was there already!

image

I would move it up though, below `zkQueries`

Comment (amonsosanz) [reply docs/web-wallet.md:227]: Regarding the content of the box, remember, these are not "Authentication Requests" but "Verification Requests", you keep using the word authentication everywhere 😅

File: sidebars.js (hunk 1 above)
Comment (amonsosanz) [reply sidebars.js:165]: Yep, I've seen it. Now it's only missing in the Build top menu.

File: docs/web-wallet.md (hunk 6 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: added

File: docs/web-wallet.md (hunk 8 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: fixed

File: docs/web-wallet.md (hunk 9 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: fixed

File: docs/web-wallet.md (hunk 10 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: changed the content

File: docs/web-wallet.md (hunk 11 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: changed the content and moved the box above

File: docs/web-wallet.md (hunk 12 above)
Comment (0xpulkit) [reply docs/web-wallet.md]: changed the content

File: docs/web-wallet.md (hunk 13 above)
Comment (0xpulkit) [reply docs/web-wallet.md:192]: added

File: sidebars.js (hunk 1 above)
Comment (amonsosanz) [reply sidebars.js:165]: > Should we add it in both- tools and build?

I'd say yes.

  * In Build, we add a link to the internal docs page, as we have on the left sidebar (we want to keep the left sidebar and the Build menu synced). T
  * In Tools we add an external link to the tool, as we do with every other web tool we have.

File: docs/web-wallet.md [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+          "https://raw.githubusercontent.com/anima-protocol/claims-polygonid/main/schemas/json-ld/pol-v1.json-ld",
+        type: "AnimaProofOfLife",
+        credentialSubject: {
+          human: {
+            $eq: true,
+          },
+        },
+      },
+    },
+  ],
```
Comment (amonsosanz) [inline docs/web-wallet.md:203]: We're lacking `verifierDid` here

File: sidebars.js (hunk 1 above)
Comment (0xpulkit) [reply sidebars.js:165]: True

File: docs/web-wallet.md (hunk 14 above)
Comment (0xpulkit) [reply docs/web-wallet.md:203]: Somehow i was in the apprehension that we don't include verifierDID with on-chain but yes after testing i got to know that we do need it as well. Let me add it

Comment (0xpulkit) [reply docs/web-wallet.md:203]: Added it!

**Review by amonsosanz — APPROVED** [review]
